device_calendar/example/lib/presentation/pages/calendar_add.dart

```diff
@@ -0,0 +1,149 @@
+import 'dart:io';
+
+import 'package:device_calendar/device_calendar.dart';
+import 'package:flutter/material.dart';
+
+class CalendarAddPage extends StatefulWidget {
+  @override
+  _CalendarAddPageState createState() {
+    return _CalendarAddPageState();
+  }
+}
+
+class _CalendarAddPageState extends State<CalendarAddPage> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+  DeviceCalendarPlugin _deviceCalendarPlugin;
+
+  bool _autovalidate = false;
+  String _calendarName = '';
+  ColorChoice _colorChoice;
+  String _localAccountName = '';
+
+  _CalendarAddPageState() {
+    _deviceCalendarPlugin = DeviceCalendarPlugin();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      key: _scaffoldKey,
+      appBar: AppBar(
+        title: Text('Create Calendar'),
+      ),
+      body: Form(
+        autovalidate: _autovalidate,
+        key: _formKey,
+        child: Container(
+          padding: EdgeInsets.all(10),
+          child: Column(
+            children: [
+              TextFormField(
+                decoration: const InputDecoration(
+                  labelText: 'Calendar Name',
+                  hintText: 'My New Calendar',
+                ),
+                validator: _validateCalendarName,
+                onSaved: (String value) => _calendarName = value,
+              ),
+              SizedBox(height: 10),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Text('Calendar Color'),
+                  DropdownButton<ColorChoice>(
+                    onChanged: (selectedColor) {
+                      setState(() => _colorChoice = selectedColor);
+                    },
+                    value: _colorChoice,
+                    items: ColorChoice.values
+                        .map((color) => DropdownMenuItem(
+                              value: color,
+                              child: Text(color.toString().split('.').last),
+                            ))
+                        .toList(),
+                  ),
+                ],
+              ),
+              if (Platform.isAndroid) ...[
```

Comment:
I don't think the ...[ is required here is it?

Reply:
argh I need to get a habit of not using the array block `...[]` for a single widget. I'll remove this in the next PR